src/corefx/System.Globalization.Native/timeZoneInfo.cpp

```diff
@@ -0,0 +1,24 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#include <stdint.h>
+#include <unistd.h>
+
+/*
+Function:
+ReadLink
+
+Gets the symlink value for the path.
+*/
+extern "C" int32_t ReadLink(const char* path, char* result, size_t resultCapacity)
+{
+	ssize_t r = readlink(path, result, resultCapacity - 1); // subtract one to make room for the NULL character
+
+	if (r < 0)
```

Comment:
should we just check if r<1 instead? returning empty string is same as null (as you checking in the managed side). so with empty string we'll avoid executing the code (like calling GetStringAndRelease).